JSDemos/Demos/Tabs/Overview/React/App.tsx

```diff
@@ -43,70 +60,99 @@ const App = () => {
     setIconPosition(e.value);
   }, [setIconPosition]);
 
-  const orientationChanged = React.useCallback(
-    (e) => {
-      setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
-      setOrientation(e.value);
-    }, [setOrientation, setWidgetWrapperClasses],
-  );
+  const orientationChanged = React.useCallback((e) => {
+    setWidgetWrapperClasses(`widget-wrapper widget-wrapper-${e.value}`);
+    setOrientation(e.value);
+  }, [setOrientation, setWidgetWrapperClasses]);
 
   const showNavigationChanged = React.useCallback((e) => {
+    const shouldRestrictWidth = e.value || scrollContent;
+
+    toggleShouldRestrictWidth(shouldRestrictWidth);
     setShowNavigation(e.value);
-  }, [setShowNavigation]);
+  }, [scrollContent, setShowNavigation, toggleShouldRestrictWidth]);
```

Comment:
The name 'toggleShouldRestrictWidth' seems a bit convoluted and potentially confusing.

'Toggle' implies switching between two states, but 'RestrictWidth' doesn't inherently suggest what those states might be. It's not immediately clear what restricting width means in this context.

`toggleWidthLimitation`, 'enforceWidthConstraint' are more clear for me.

Reply:
[5b9bd3875eddb26acdca83ce94cfe7a767786467]